test/src/main/java/org/apache/accumulo/test/TableOperationsIT.java

```diff
@@ -202,6 +205,64 @@ public void createTable() throws TableExistsException, AccumuloException,
     accumuloClient.tableOperations().delete(tableName);
   }
 
+  @Test
+  public void testDefendAgainstThreadsCreateSameTableNameConcurrently()
+      throws ExecutionException, InterruptedException {
+
+    final int numTasks = 10;
+    ExecutorService pool = Executors.newFixedThreadPool(numTasks);
+
+    for (String tablename : getUniqueNames(30)) {
+      CountDownLatch startSignal = new CountDownLatch(1);
+      CountDownLatch doneSignal = new CountDownLatch(numTasks);
+      AtomicInteger numTasksRunning = new AtomicInteger(0);
+
+      List<Future<Boolean>> futureList = new ArrayList<>();
+
+      for (int j = 0; j < numTasks; j++) {
+        Future<Boolean> future = pool.submit(() -> {
+          boolean result;
+          try {
+            numTasksRunning.incrementAndGet();
+            startSignal.await();
+            accumuloClient.tableOperations().create(tablename);
+            result = true;
+          } catch (TableExistsException e) {
+            result = false;
+          } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            result = false;
+          } finally {
+            doneSignal.countDown();
+          }
+          return result;
+        });
+        futureList.add(future);
+      }
+
+      Wait.waitFor(() -> numTasksRunning.get() == numTasks);
+
+      startSignal.countDown();
+
+      doneSignal.await();
+
+      int taskSucceeded = 0;
+      int taskFailed = 0;
+      for (Future<Boolean> result : futureList) {
+        if (result.get() == true) {
+          taskSucceeded++;
+        } else {
+          taskFailed++;
+        }
+      }
+
+      assertEquals(1, taskSucceeded);
+      assertEquals(9, taskFailed);
+    }
+
```

Comment:
Would be good to call `accumuloClient.tableOperations().list()` and ensure the returned set contains all of the tables created.